torch/distributed/_composable/checkpoint_activation.py

```diff
@@ -1,15 +1,19 @@
 from contextlib import contextmanager
 from functools import partial
-from typing import Any, List, Optional, Tuple
+from typing import Any, Callable, Dict, List, Optional, Tuple
 from weakref import ref, ReferenceType, WeakKeyDictionary
 
 import torch
 import torch.nn as nn
+from torch.distributed.utils import _pack_kwargs, _unpack_kwargs
 from torch.utils.checkpoint import detach_variable, get_device_states, set_device_states
 
 from .contract import contract
 
 
+__all__ = ["checkpoint"]
+
+
 @contextmanager
 def _no_hook(module: nn.Module):
     r"""
```

Comment:
This is a pre-existing issue, but since I was recently fixing an issue related to this: is `enable_hook` thread local? Because if it isn't, seems like you could end up in a threading race condition with multi-device backwards.